src/index.ts

```diff
@@ -142,6 +159,30 @@ module.exports = class ServerlessPlugin {
 
     this.addTags(handlers, config.enableTags);
 
+    const simpleGit = await newSimpleGit();
+
+    // We only want to enable source code integration if the following conditions are met:
+    // 1. The flag is enabled (it's enabled by default)
+    // 2. They've provided a DATADOG_API_KEY (this will silently fail if they do not provide an API key)
+    // 3. They're in a Git repo (this will silently fail if they do not provide an API key)
```

Comment:
I don't think you need this comment here. If you think the code isn't clear by itself (I think it's fine as is) you could make it more obvious by doing something like:

```
const sourceCodeIntegrationEnabled = ...
const apiKeyProvided = ...
const isGitRepo = ...

if (sourceCodeIntegrationEnabled && apiKeyProvided && isGitRepo) {
...
```